libkpod/ps.go

```diff
@@ -0,0 +1,192 @@
+package libkpod
+
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/kubernetes-incubator/cri-o/oci"
+	"github.com/pkg/errors"
+)
+
+// FilterParamsPS contains the filter options for ps
+type FilterParamsPS struct {
+	id       string
+	label    string
+	name     string
+	exited   int32
+	status   string
+	ancestor string
+	before   time.Time
+	since    time.Time
+	volume   string
+}
+
+// ParseFilter takes a filter string and a list of containers and filters it
+func ParseFilter(filter string, server *ContainerServer) (*FilterParamsPS, error) {
```

Comment:
Can make the list of containers a parameter, instead of ContainerServer, to make this possible.